internal/config/nsmgr/nsmgr.go

```diff
@@ -187,6 +187,50 @@ func (mgr *NamespaceManager) dirForType(ns NSType) string {
 	return filepath.Join(mgr.namespacesDir, string(ns)+"ns")
 }
 
+// NamespaceFromProcEntry creates a new namespace object from a bind mount from a processes proc entry.
+// The caller is responsible for cleaning up the namespace by calling Namespace.Remove().
+// This function is heavily based on containernetworking ns package found at:
+// https://github.com/containernetworking/plugins/blob/5c3c17164270150467498a32c71436c7cd5501be/pkg/ns/ns.go#L140
+// Credit goes to the CNI authors.
+func (mgr *NamespaceManager) NamespaceFromProcEntry(pid int, nsType NSType) (_ Namespace, retErr error) {
+	pinnedNamespace := uuid.New().String()
+
+	// now create an empty file
+	f, err := os.Create(pinnedNamespace)
```

Comment:
Perhaps it's better to use `os.CreateTemp` instead -- it actually checks that the file being created has a unique name (rather than relying on UUID being unique), which seems more straightforward.

I'm also a bit concerned about _where_ the file is being created. I guess the code assumes we're in a particular directory -- maybe add a comment about it.

Reply:
ooh great catch, it wasn't putting it in the right directory

Reply:
haha I guess I figured it out eventually